### PR TITLE
feat: dns validations

### PR DESCRIPTION
**Pull request by cbrea**
1 - Implemented a configurable DNS response timeout

2 - Draft to avoid eDNS0 OPT record values as SHA256

### CODE REVIEW

File: steps/dns/steps.go [hunk 1]
```diff
@@ -38,8 +40,17 @@ func (s Steps) InitializeSteps(ctx context.Context, scenCtx *godog.ScenarioConte
 	ctx = InitializeContext(ctx)
 	session := GetSession(ctx)
 
+	// Set the DNS responses timeout
+	// Set if indicated in environment parameter: CONF:timeout
+	// If not indicated, the default value is kept in 2 seconds (mikeg library)
+
+	if timeout, err := strconv.ParseInt(golium.ValueAsString(ctx, "[CONF:timeout]"), 10, 64); err == nil {
```
Comment (jlorgal) [inline steps/dns/steps.go:47]: I guess that it's more natural to use a step to configure the timeout (I guess that it's not a usual step that you require).
Something like: 
`^a DNS timeout of "(\d+)" milliseconds$`

You are linking the steps with a specific configuration setting, that I don't think it's recommendable.

Other option is to including these settings, as default values, in the Steps struct. However, I would recommend the simple approach of creating another step to configure the timeout.

File: steps/dns/steps.go [hunk 2]
```diff
@@ -38,8 +40,17 @@ func (s Steps) InitializeSteps(ctx context.Context, scenCtx *godog.ScenarioConte
...
+		session.Timeout = time.Duration(timeout) * time.Millisecond
+	}
+
 	// Initialize the steps
 	scenCtx.Step(`^the DNS server "([^"]*)"$`, func(svr string) error {
+
```
Comment (jlorgal) [inline steps/dns/steps.go:53]: remove this empty line

File: test/acceptance/environments/local.yml [hunk 3]
```diff
@@ -4,3 +4,4 @@ host: httpbin.org
 
 # DNS settings
 dns: 8.8.8.8:53
+timeout: "2000"
```
Comment (jlorgal) [inline test/acceptance/environments/local.yml:7]: Why do you use a string? Why not `timeout: 2000`?

**Review by jlorgal — CHANGES_REQUESTED** [review]

File: test/acceptance/environments/local.yml (hunk 3 above)
Comment (cbrea) [reply test/acceptance/environments/local.yml]: Due value.go restriction in CONF tagname.  I've solved it adding a %v verb
`Error: Error casting timeout. strconv.Atoi: parsing "%!s(float64=2000)": invalid syntax`

File: steps/dns/steps.go [hunk 4]
```diff
@@ -98,7 +106,8 @@ func parseOptionsTable(ctx context.Context, t *godog.Table) ([]dns.EDNS0, error)
 	for i, o := range options {
 		data, err := hex.DecodeString(o.Data)
 		if err != nil {
-			return nil, fmt.Errorf("Error converting to byte array: %s. %s", o.Data, err)
+			//return nil, fmt.Errorf("Error converting to byte array: %s. %s", o.Data, err)
+			data = []byte(o.Data)
```
Comment (jlorgal) [inline steps/dns/steps.go:110]: Why do you do that?

Comment (cbrea) [reply steps/dns/steps.go:110]: I'd like to remove this restriction to allow any value in OPT records. Currently only SHA256 values are allowed. F.I:

   ```
   And the DNS query options
          | code | data                 |
          | 1001 | Incorrect-Value |
          | 1002 | [EMPTY]           |
```

**Review by jlorgal — APPROVED** [review]
LGTM

**Pull request merged**